docs/design-docs/reporting-sot.md

```diff
@@ -0,0 +1,538 @@
+viable market value column. Removing the 95% CI here since it's largely unused.
+
+- `pin_n_tot`
+- `pin_n_w_value`
+- `pin_pct_w_value`
+- `reassessment_year`
+- `sale_n_tot`
+- `sale_n_valid_ccao` (CCAO sales validation codes)
+- `sale_n_invalid_ccao`
+- `sale_n_valid_idor` (IDOR sales validation codes)
+- `sale_n_invalid_idor`
+- `fmv_tot_min`
+- `fmv_tot_q10`
+- `fmv_tot_q25`
+- `fmv_tot_median`
+- `fmv_tot_q75`
+- `fmv_tot_q90`
+- `fmv_tot_max`
+- `fmv_tot_mean`
+- `fmv_tot_sum`
+- `fmv_tot_delta_median`
+- `fmv_tot_delta_mean`
+- `fmv_tot_delta_sum`
+- `fmv_tot_delta_pct_median`
+- `fmv_tot_delta_pct_mean`
+- `ratio_min`
+- `ratio_q10`
+- `ratio_q25`
+- `ratio_median`
+- `ratio_q75`
+- `ratio_q90`
+- `ratio_max`
+- `ratio_mean`
+- `cod`
+- `cod_no_sop`
+- `cod_n`
+- `prd`
+- `prd_no_sop`
+- `prd_n`
+- `prb`
+- `prb_no_sop`
+- `prb_n`
+- `mki`
+- `mki_no_sop`
+- `mki_n`
+- `ratio_met`
+- `cod_met`
+- `prd_met`
+- `prb_met`
+- `mki_met`
+- `vertical_equity_met`
+- `within_05_pct`
+- `within_10_pct`
+- `within_15_pct`
+- `within_20_pct`
+
+### Implementation
+
+- Tables should be constructed using our newly implemented dbt Python model
+  support. Preference to use PySpark for speed of construction, Pandas as
+  a fallback
+- Tables should be materialized as tables, i.e. saved as Parquet files. It may
+  be worth experimenting with file partitioning, assuming the files generated
+  are sufficiently large
+- Tests and documentation should be defined as usual, in the `schema.yml` file
+  of each dbt model
+- Table should use a new prefix `sot_` in their names
+- Tables should be constructed such that adding new columns and/or groups is
+  easy and non-breaking
+- Major changes to the tables, such as changes to ratio statistic or sales
+  validation methodologies, should be accompanied by incrementing the `method`
+  column (e.g. `v1` becomes `v2`). The prior method version should *only be
+  deprecated with the consent of all major users*. Method changes should be
+  documented in the table documentation
+
+### Testing
+
+All new tables should be accompanied by dbt data tests (and unit tests where
+appropriate). Data sets should ensure the following *at minimum*:
+
+- Geographies and groups contain their representative sets, i.e. the
+  `municipality` geography contains all county municipalities, and `class`
+  contains all CCAO class codes
+- Stages contain their representative sets, i.e. `mailed`, `certified`, and
+  `board` stages exist for each year regardless of whether or not they've
+  been finalized
+- Values fall within expected ranges
+  - Counts are always greater than or equal to 0
+  - FMVs and AVs are constrained by
+- No null, inf, or NaN values are present (where appropriate, some nulls are
+  expected e.g. not all PINs are covered by some geography types)
+- Tables are unique by primary key
+- Tests are run daily (as part of the `build-daily-dbt-models` workflow)
```

Comment:
**[Thought, non-blocking]** Nice tests! Just flagging that we don't currently have a system for running non-iasWorld data tests on a set schedule, so this will likely involve coming up with a new tag and then updating `build-daily-dbt-models` to run tests with that tag -- I think that's what this section is suggesting too, but I just want to make it explicit.

Reply:
In addition to things like municipality containing all county municipalities, let's avoid confirmation bias and be sure to test the opposite: that there are no municipalities that are not in the list of county municipalities, or wards that are not real wards (e.g., Evanston ward 30). I suspect we already have tests for this but just signal-boosting!